app/core/treema-ext.coffee

```diff
@@ -398,6 +399,36 @@ module.exports.LatestVersionReferenceNode = class LatestVersionReferenceNode ext
     return 'Unknown - ' + (data.original ? data) unless m
     return @modelToString(m)
 
+  getReferencedModel: (data, schema) ->
+    return null unless schema.links?
+    linkObject = _.find schema.links, rel: 'db'
+    return null unless linkObject
+    return null if linkObject.href.match('thang.type') and not CocoModel.isObjectID(data)  # Skip loading hardcoded Thang Types for now (TODO)
+
+    # not fully extensible, but we can worry about that later
+    link = linkObject.href
+    link = link.replace('{(original)}', data.original)
+    link = link.replace('{(majorVersion)}', '' + (data.majorVersion ? 0))
+    link = link.replace('{($)}', data)
+    @getOrMakeModelFromLink(link)
+
+  getOrMakeModelFromLink: (link) ->
+    makeUrlFunc = (url) -> -> url
+    modelUrl = link.split('/')[2]
+    modelModule = _.string.classify(modelUrl)
+    modulePath = "models/#{modelModule}"
+
+    modulePath = modulePath.replace(/^models\//,'')
+    try
+      Model = require('app/models/' + modulePath) # TODO webpack: Get this working async for chunking
+    catch e
+      console.error 'could not load model from link path', link, 'using path', modulePath
+      return
+
+    model = new Model()
+    model.url = makeUrlFunc(link)
+    return model
+
```

Comment:
👍 
Moved these here to make CocoModel not dependent on every model ever (as happens on line 423).
Things that use Treema are dependent on every model, but that's mostly Editor pages, which I think makes this fine.